hash_practice/exercises.py

```diff
@@ -1,19 +1,52 @@
 
+def create_anagram_key(string):
+    return ''.join(sorted(string))
+
+
 def grouped_anagrams(strings):
     """ This method will return an array of arrays.
         Each subarray will have strings which are anagrams of each other
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(n*m)
+        Space Complexity: O(n)
     """
-    pass
+    grouped_words = {}
+    anagrams = []
+
+    for ele in strings:
+        if grouped_words.get(create_anagram_key(ele)) == None:
+            grouped_words[create_anagram_key(ele)] = [ele]
+        else:
+            grouped_words[create_anagram_key(ele)].append(ele)
+
+    for k, v in grouped_words.items():
+        anagrams.append(v)
+
+    return anagrams
+
 
 def top_k_frequent_elements(nums, k):
     """ This method will return the k most common elements
         In the case of a tie it will select the first occuring element.
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(n)

```

Comment:
Don't forget to account for the call to `sort` on line 42, which will take O(n * log(n)) time and thus be the dominant part of the algorithm for the final time complexity.